dataset/src/main/kotlin/com/cosmotech/dataset/azure/DatasetServiceImpl.kt

```diff
@@ -159,6 +161,32 @@ internal class DatasetServiceImpl(
     TODO("Not yet implemented")
   }
 
+  override fun searchDatasets(organizationId: String, datasetSearch: DatasetSearch): List<Dataset> {
+    var searchList: Set<Dataset> = setOf()
+    if (datasetSearch.datasetTags == null) {
```

Comment:
Since we expect `datasetTags` to be filled, I suggest we mark it as `required` right in the OpenAPI definition, so a non-nullable Kotlin type is generated.

Reply:
done